ast-nodes/routine-call.dart

```diff
@@ -9,13 +12,48 @@ class RoutineCall implements Primary {
 
   RoutineCall(this.name, this.arguments);
 
-  // TODO: implement .parse()
+  factory RoutineCall.parse(Iterable<Token> tokens) {
+    final iter = tokens.iterator;
+    checkNext(iter, RegExp('[a-zA-Z_]\w*\$'), "Expected identifier");
+    final tempName = iter.current.value;
+    checkNext(iter, RegExp('^[(]\$'), 'Expected "("');
+    var exprs = []; // Final array of parsed Expressions
+    var expBuffer = []; // Array of Tokens that are located between main braces
+    var argExpected =
+        false; // Flag that is set after encountering a comma (to detect missing arguments)
+    String
+        lastToken; // Store the last Token in the expBuffer in order to prevent trailing comma
+    expBuffer = consumeStackUntil(iter, RegExp('^[(]\$'), RegExp('^[)]\$'));
+    if (iter.current?.value != ')') {
+      throw SyntaxError(null, 'Expected ")"');
+    }
+    final exprIter = expBuffer.iterator;
+    while (exprIter.moveNext()) {
+      final tokenBuff = consumeUntil(exprIter, RegExp("^[,]\$"));
+      // Check for the case with missing argument
+      if (tokenBuff.isEmpty && argExpected) {
+        throw SyntaxError(null, 'Expected an argument');
+      } else if (!tokenBuff.isEmpty) {
+        exprs.add(Expression.parse(tokenBuff));
+      }
+      if (exprIter.current?.value == ',') {
+        argExpected = true;
+      }
+      lastToken = exprIter.current?.value;
+    }
+    // Check for the trailing comma
+    if (lastToken == ',') {
+      throw SyntaxError(null, 'Expected an argument');
+    }
+    return RoutineCall(tempName, exprs.isEmpty ? null : exprs);
```

Comment:
Better use `[]` instead of `null`, because the `toString` method breaks on `null`.